File: src/entities/countryPrices.js
/**
 * @class Country
 */
export default class CountryPrices {
    /**
     * @param {string} name
     * @param {string} code
     */
    constructor(name, code) {
        /** @member {string} CountryPrices#name */
        this.name = name;

        /** @member {string} CountryPrices#code */
        this.code = code;

        /** @member {double|undefined} CountryPrices#dieselPrice */
        this.dieselPrice = 0;

        /** @member {double|undefined} CountryPrices#electricityPrice */
        this.electricityPrice = 0;

        /** @member {double|undefined} CountryPrices#oneTimeSubsidy */
        this.oneTimeSubsidy = 0;

        /** @member {double|undefined} CountryPrices#annualSubsidy */
        this.annualSubsidy = 0;
    }
}

File: server/countryPrices.py
import csv


class CountryPrices:
    def __init__(self, name, code):
        self.name = name
        self.code = code
        self.dieselPrice = 0.0
        self.electricityPrice = 0.0
        self.oneTimeSubsidy = 0.0
        self.annualSubsidy = 0.0


class CountryPricesService:
    def __init__(self):
        pass

    @staticmethod
    def get_all():
        with open('./data/country-prices.csv', 'rb') as f:
            reader = csv.reader(f)
            data_list = list(reader)

            return filter(lambda item: item is not None,
                          map(CountryPricesService.create_entity_from_csv_row, data_list))

    @staticmethod
    def create_entity_from_csv_row(row):
        try:
            entity = CountryPrices(row[0], row[1])
            entity.dieselPrice = float(row[2])
            entity.electricityPrice = float(row[3])
            entity.oneTimeSubsidy = float(row[4])
            entity.annualSubsidy = float(row[5])
            return entity
        except:
            return None
